pkg/ui/src/views/statements/planView.tsx

```diff
@@ -16,115 +16,314 @@ import _ from "lodash";
+      }
+      return <div className="arrow">{DOWN_ARROW_ICON}</div>;
     }
-    const existingAttr = collapsed[attr.key];
-    if (!existingAttr) {
-      collapsed[attr.key] = {
-        key: attr.key,
-        value: [attr.value],
-      };
-      return;
-    } else {
-      collapsed[attr.key].value.push(attr.value);
+  }
+
+  renderMaybeExpandedDetail() {
+    const node = this.props.node;
+    if (node.attrs && node.attrs.length > 0) {
+      if (this.state.expanded) {
+        return (
+          <div className="nodeAttributes">
+            {node.attrs.map( (attr) => (
+              <div key={attr.key} className="attr">
+                {attr.key}
+                {attr.value &&
+                  <span>={attr.value}</span>
+                }
+              </div>
+            ))}
+          </div>
+        );
+      }
+    }
+  }
+
+  renderClassName(warn: boolean) {
+    const node = this.props.node;
+    if (node.attrs && node.attrs.length > 0 && this.state.expanded) {
+      if (warn) {
+        return "nodeDetails warn expanded";
+      }
+      return "nodeDetails expanded";
+    } else if (warn) {
+      return "nodeDetails warn";
     }
-  });
+    return "nodeDetails";
+  }
+
+  render() {
+    const node = this.props.node;
+    const headerProps = planNodeHeaderProps(node);
+    return (
+      <div
+        className={this.renderClassName(headerProps.warn)}
+        onClick={() => this.toggleExpanded()}
+      >
+        {this.renderPlanNodeHeader(headerProps)} {this.renderArrow()}
+        {this.renderMaybeExpandedDetail()}
+      </div>
+    );
+  }
+}
+
+interface PlanNodeProps {
+  node: FlatPlanNode;
+}
+
+class PlanNode extends React.Component<PlanNodeProps> {
+  render() {
+    const node = this.props.node;
+    return <li>
+      <PlanNodeDetails
+        node={node}
+      />
+      {node.children && node.children.map( (child) => (
+        <PlanNodes
+          nodes={child}
+        />
+      ))
+      }
+    </li>;
+  }
+}
+
+function PlanNodes(props: {
+  nodes: FlatPlanNode[],
+}): React.ReactElement<{}> {
+  const nodes = props.nodes;
+  return <ul>
+    {nodes.map( (node) => {
+      return <PlanNode node={node}/>;
+    })}
+  </ul>;
+}
+
+export function PlanView(props: {
+  title: string,
+  plan: IExplainTreePlanNode,
+}) {
+  const flattenedPlanNodes = flattenTree(props.plan);
+
+  const lastSampledHelpText = (
+    <React.Fragment>
+      If the time from the last sample is greater than 5 minutes, a new
+      plan will be sampled. This frequency can be configured
+      with the cluster setting{" "}
+      <code>
+        <pre style={{ display: "inline-block" }}>
+          sql.metrics.statement_details.plan_collection.period
+        </pre>
+      </code>.
+    </React.Fragment>
+  );
 
-  const collapsedAttrs = _.values(collapsed);
-  return _.sortBy(collapsedAttrs, (ca) => (
-    ca.key === "table" ? "table" : "z" + ca.key
-  ));
+  return <table className="plan-view-table">
```

Comment:
style nit: use parens

```
return (
  ...
);
```

Reply:
The rationale here is to always make open and close tags line up. Prettier does this automatically.